packages/ia-topnav/src/ia-topnav.js

```diff
@@ -0,0 +1,237 @@
+  }
+
+  signedOutMenu() {
+    this.userMenuOpen = false;
+    this.searchMenuOpen = false;
+    this.mediaMenuOpen = false;
+    this.signedOutMenuOpen = !this.signedOutMenuOpen;
+  }
+
+  searchMenu() {
+    this.userMenuOpen = false;
+    this.mediaMenuOpen = false;
+    this.signedOutMenuOpen = false;
+    this.searchMenuAnimate = true;
+    this.searchMenuFade = true;
+    this.searchMenuOpen = !this.searchMenuOpen;
+    if (this.searchMenuOpen) {
+      window.setTimeout(() => {
+        this
+          .shadowRoot
+          .querySelector('primary-nav')
+          .shadowRoot
+          .querySelector('nav-search')
+          .shadowRoot
+          .querySelector('.search-field').focus();
+      }, 0);
+    }
+  }
+
+  userMenu() {
+    this.searchMenuOpen = false;
+    this.mediaMenuOpen = false;
+    this.signedOutMenuOpen = false;
+    this.userMenuAnimate = true;
+    this.userMenuOpen = !this.userMenuOpen;
+  }
+
+  closeMenus() {
+    this.mediaMenuOpen = false;
+    this.mediaSliderOpen = false;
+    this.searchMenuOpen = false;
+    this.signedOutMenuOpen = false;
+    this.userMenuOpen = false;
+    this.selectedMenuOption = '';
+  }
+
+  navSearch(e) {
+    this.searchSubmitted = true;
+    const { originalEvent, formEl } = e.detail;
+    const searchIn = document
+      .querySelector('ia-topnav')
+      .shadowRoot
+      .querySelector('search-menu')
+      .shadowRoot
+      .querySelector('[name=sin]:checked')
+      .value;
+    const query = formEl.querySelector('[name=query]').value;
+    if (!query) {
+      originalEvent.preventDefault();
+      this.searchSubmitted = false;
+      return false;
+    }
+    formEl.querySelector('[name=sin]').value = searchIn;
+    return true;
+  }
+
+  trackClick({ detail }) {
+    this.dispatchEvent(new CustomEvent('analyticsClick', {
+      bubbles: true,
+      composed: true,
+      detail,
+    }));
+  }
+
+  trackSubmit({ detail }) {
+    this.dispatchEvent(new CustomEvent('analyticsSubmit', {
+      bubbles: true,
+      composed: true,
+      detail,
+    }));
+  }
+
+  mediaTypeSelected({ detail }) {
+    if (this.selectedMenuOption === detail.mediatype) {
+      this.closeMediaSlider();
+      return;
+    }
+    this.selectedMenuOption = detail.mediatype;
+    this.toggleMediaSlider();
+  }
+
+  get isMenuOpen() {
+    return this.mediaMenuOpen
+      || this.searchMenuOpen
+      || this.userMenuOpen
+      || this.mediaSliderOpen
+      || this.signedOutMenuOpen;
+  }
+
+  render() {
+    const searchMenuTabIndex = this.searchMenuOpen ? '' : '-1';
+    const userMenuTabIndex = this.userMenuOpen ? '' : '-1';
+    const closeLayerVisible = this.isMenuOpen;
+
+    return html`
+      <div class='topnav'>
+        <primary-nav
+          .config=${this.config}
+          .selectedMenuOption=${this.selectedMenuOption}
+          .mediaMenuOpen="${this.mediaMenuOpen}"
+          .searchMenuFade="${this.searchMenuFade}"
+          .searchMenuOpen="${this.searchMenuOpen}"
```

Comment:
What do these `*Open` properties do?

Reply:
Determine opened state of various menus.